File: levelledmobs-plugin-bukkit/src/main/java/io/github/arcaneplugins/levelledmobs/bukkit/logic/group/GroupPreParseEvent.java
package io.github.arcaneplugins.levelledmobs.bukkit.logic.group;

import org.bukkit.event.Cancellable;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;
import org.jetbrains.annotations.NotNull;

public final class GroupPreParseEvent extends Event implements Cancellable {

    /* vars */

    private static final HandlerList HANDLERS = new HandlerList();
    private final Group group;
    private boolean cancelled = false;

    /* constructors */

    public GroupPreParseEvent(final @NotNull Group group) {
        this.group = group;
    }

    /* getters and setters */

    @NotNull
    public Group getGroup() { return group; }

    @Override
    public boolean isCancelled() {
        return cancelled;
    }

    @Override
    public void setCancelled(boolean state) {
        this.cancelled = state;
    }

    @NotNull
    @Override
    public HandlerList getHandlers() {
        return HANDLERS;
    }

    @NotNull
    public static HandlerList getHandlerList() { return HANDLERS; }
}

File: levelledmobs-plugin-bukkit/src/main/kotlin/io/github/arcaneplugins/levelledmobs/bukkit/logic/group/GroupPreParseEvent.kt
package io.github.arcaneplugins.levelledmobs.bukkit.logic.group

import org.bukkit.event.Cancellable
import org.bukkit.event.Event
import org.bukkit.event.HandlerList


class GroupPreParseEvent(val group: Group) : Event(), Cancellable {
    companion object {
        val HANDLERS = HandlerList()
    }

    private var cancelled = false

    override fun isCancelled(): Boolean {
        return this.cancelled
    }

    override fun setCancelled(state: Boolean){
        this.cancelled = state
    }

    override fun getHandlers(): HandlerList {
        return HANDLERS;
    }
}